torch_geometric/explain/algorithm/subgraphX_explainer.py

```diff
@@ -0,0 +1,646 @@
+                          edge_index=remove_self_loops(self.edge_index)[0])
+        self.graph = to_networkx(graph_data, to_undirected=True)
+        self.data = Batch.from_data_list([self.data])
+        self.num_nodes = self.graph.number_of_nodes()
+        self.score_func = score_func
+        self.n_rollout = n_rollout
+        self.min_atoms = min_atoms
+        self.c_puct = c_puct
+        self.expand_atoms = expand_atoms
+        self.high2low = high2low
+        self.new_node_idx = None
+
+        # extract the sub-graph and change the node indices.
+        if node_idx is not None:
+            if isinstance(node_idx, torch.Tensor):
+                node_idx = node_idx.item()
+
+            self.ori_node_idx = node_idx
+            self.ori_graph = copy.copy(self.graph)
+            x, edge_index, subset, edge_mask, kwargs = self.__subgraph__(
+                node_idx, self.X, self.edge_index, self.num_hops)
+            self.data = Batch.from_data_list(
+                [Data(x=x, edge_index=edge_index)])
+            self.graph = self.ori_graph.subgraph(subset.tolist())
+            mapping = {int(v): k for k, v in enumerate(subset)}
+            self.graph = nx.relabel_nodes(self.graph, mapping)
+            self.new_node_idx = torch.where(
+                subset == self.ori_node_idx)[0].item()
+            self.num_nodes = self.graph.number_of_nodes()
+            self.subset = subset
+
+        self.root_coalition = sorted([node for node in range(self.num_nodes)])
+        self.MCTSNodeClass = partial(
+            MCTSNode,
+            data=self.data,
+            ori_graph=self.graph,
+            c_puct=self.c_puct,
+            device=self.device,
+        )
+        self.root = self.MCTSNodeClass(self.root_coalition)
+        self.state_map = {str(self.root.coalition): self.root}
+
+    def set_score_func(self, score_func):
+        self.score_func = score_func
+
+    @staticmethod
+    def __subgraph__(node_idx, x, edge_index, num_hops, **kwargs):
+        num_nodes, num_edges = x.size(0), edge_index.size(1)
+        subset, edge_index, _, edge_mask = k_hop_subgraph(
+            node_idx, num_hops, edge_index, relabel_nodes=True,
+            num_nodes=num_nodes)
+
+        x = x[subset]
+        for key, item in kwargs.items():
+            if torch.is_tensor(item) and item.size(0) == num_nodes:
+                item = item[subset]
+            elif torch.is_tensor(item) and item.size(0) == num_edges:
+                item = item[edge_mask]
+            kwargs[key] = item
+
+        return x, edge_index, subset, edge_mask, kwargs
+
+    def mcts_rollout(self, tree_node):
+        cur_graph_coalition = tree_node.coalition
+        if len(cur_graph_coalition) <= self.min_atoms:
+            return tree_node.P
+
+        # Expand if this node has never been visited
+        if len(tree_node.children) == 0:
+            node_degree_list = list(
+                self.graph.subgraph(cur_graph_coalition).degree)
```

Comment:
Could you store the degrees of all of the nodes initially and then just take a slice of the array to get the candidate choices? Also, is there a reason you are using self.graph vs. x and edge_index.